### PR TITLE
Add campaign experiment example

### PR DESCRIPTION
**Pull request by nwbirnie**

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+   * result.
+   */
+  private CreateCampaignExperimentMetadata createExperiment(
+      GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId, long draftId) {
+    // Defines the experiment to be created.
+    CampaignExperiment experiment =
+        CampaignExperiment.newBuilder()
+            .setCampaignDraft(
+                StringValue.of(ResourceNames.campaignDraft(customerId, baseCampaignId, draftId)))
+            .setName(StringValue.of("Campaign experiment = " + UUID.randomUUID()))
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:120]: Most of our other examples follow the pattern: "Name #" + System.currentTimeMillis()

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      // Block on the long running (i.e. async) operation result.
+      try {
+        return campaignExperimentAsync.getMetadata().get();
+      } catch (InterruptedException | ExecutionException e) {
+        throw new RuntimeException("Create operation failed", e);
+      }
+    }
+  }
+
+  /** Retrieves a CampaignExperiment and prints associated information. */
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:142]: Add params to javadoc.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  private void printExperimentDetails(
+      GoogleAdsClient googleAdsClient, long customerId, CreateCampaignExperimentMetadata metadata) {
+    // Defines a basic query to retrieve the experiment.
+    String query =
+        "SELECT campaign_experiment.experiment_campaign\n"
+            + "FROM campaign_experiment\n"
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:148]: Replace newline with space

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues a search request.
+      SearchPagedResponse response =
+          googleAdsServiceClient.search(String.valueOf(customerId), query);
+
+      // Iterates through the results and prints the information returned.
+      for (GoogleAdsRow row : response.iterateAll()) {
+        System.out.printf(
+            "Experiment campaign with resource ID = '%s' created successfully.%n",
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:163]: resource ID -> resource name

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    }
+  }
+
+  /** Runs the example. */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, long baseCampaignId, long draftId) {
+    // Creates the campaign experiment.
+    CreateCampaignExperimentMetadata metadata =
+        createExperiment(googleAdsClient, customerId, baseCampaignId, draftId);
```
Comment (devchas) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:99]: The slight misalignment of comment, method name, and return type could be confusing. Is there a way to capture both? createExperimentAndReturnMetadata seems unwieldy. At a minimum, let's update the comment here and in the javadoc of the method.

Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:99]: I think it's clear enough that we're returning metadata for the experiment... could you clarify what changes you would want to the comment/javadoc please?

Comment (devchas) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/CreateCampaignExperiment.java:107]: Right.

**Pull request merged**